app/main.py

```diff
@@ -1,68 +1,71 @@
+from __future__ import annotations
+
+from typing import Union
+
+
 class Distance:
-    def __init__(self, km: float):
+    def __init__(self, km: Union[int, float]) -> None:
         self.km = km
 
-    def __str__(self):
+    def __str__(self) -> str:
         return f"Distance: {self.km} kilometers."
 
-    def __repr__(self):
+    def __repr__(self) -> str:
         return f"Distance(km={self.km})"
 
-    def __add__(self, other):
+    def __add__(self, other: Union[Distance, int, float]) -> Distance:
         if isinstance(other, Distance):
             return Distance(self.km + other.km)
         else:
             return Distance(self.km + other)
 
-    def __iadd__(self, other):
+    def __iadd__(self, other: Union[Distance, int, float]) -> Distance:
         if isinstance(other, Distance):
             self.km += other.km
             return self
         else:
             self.km += other
             return self
 
-    def __mul__(self, other):
-        if isinstance(other, Distance):
-            return Distance(self.km * other.km)
-        else:
+    def __mul__(self, other: Union[int, float]) -> Union[Distance, None]:
+        if isinstance(other, (int, float)):
             return Distance(self.km * other)
 
-    def __truediv__(self, other):
+    def __truediv__(self, other: Union[int, float]) -> Union[Distance, None]:
         if isinstance(other, Distance):
             return Distance(round(self.km / other.km, 2))
         else:
             return Distance(round(self.km / other, 2))
 
```

Comment:
Rewrite this metod as __mul__. Other can`t be type Distance same as in mul. Beaucouse if u do this - u get km-square.